framework/src/meshgenerators/PolyLineMeshFollowingNodeSetGenerator.C

```diff
@@ -0,0 +1,211 @@
+  for (auto i : make_range(n_points))
+  {
+    // Move the point forward in the search direction
+    const auto previous_point = current_point;
+    current_point += previous_direction * _dx;
+
+    // Draw a sphere to find the next point as the barycenter of the nodes from the nodeset inside
+    // the sphere.
+    // NOTE: there are many heuristics we could use here. We could try to fit a cylinder
+    // to the nodeset if we know the nodeset represents a cylinder for example.
+    Point barycenter(0);
+    unsigned int n_sum = 0;
+    for (const auto n_id : nodeset_nodes)
+      if ((current_point - base_mesh->node_ref(n_id)).norm_sq() < search_radius_sq)
+      {
+        if (!_ignore_nodes_behind ||
+            ((base_mesh->node_ref(n_id) - current_point) * previous_direction >= 0))
+        {
+          barycenter += base_mesh->node_ref(n_id);
+          n_sum++;
+        }
+      }
+    if (n_sum > 0)
+      barycenter /= n_sum;
+    else if (!_ignore_nodes_behind)
+      mooseError("Did not find any nodes in the nodeset near the current point at: ",
+                 current_point);
+    else
+      barycenter = previous_point;
+
+    if (MooseUtils::absoluteFuzzyEqual((barycenter - previous_point).norm_sq(), 0))
+    {
+      mooseInfo("Barycenter did not move from ", barycenter, ". Returning!");
+      goto done_drawing;
+    }
+
+    // Compute new direction
+    const auto new_direction = (barycenter - previous_point).unit();
+    previous_direction = new_direction;
+
+    // Set the new point towards the barycenter
+    n_segments++;
+    // Note: this dx could be different than the dx used to search the barycenter
+    current_point = previous_point + _dx * new_direction;
+    mesh.add_point(current_point, (i + 1) * _num_edges_between_points);
+    if (_verbose)
+      _console << i << ": new point: " << current_point << " new direction " << previous_direction
+               << std::endl;
+
+    // Add the additional edges in between if requested
+    if (_num_edges_between_points > 1)
+    {
+      auto p = previous_point;
+      const Point pvec = (current_point - previous_point) / _num_edges_between_points;
+      for (auto j : make_range(1u, _num_edges_between_points))
+      {
+        p += pvec;
+        mesh.add_point(p, i * _num_edges_between_points + j);
+      }
+    }
+  }
+
+done_drawing:
+
+  const auto n_elem = n_segments * _num_edges_between_points - 1;
+  const auto max_nodes = n_segments * _num_edges_between_points - 1;
+  for (auto i : make_range(n_elem + _loop))
+  {
+    const auto ip1 = _loop ? (i + 1) % max_nodes : (i + 1);
+    auto elem = Elem::build(EDGE2);
+    elem->set_node(0, mesh.node_ptr(i));
+    elem->set_node(1, mesh.node_ptr(ip1));
+    elem->set_id() = i;
+    mesh.add_elem(std::move(elem));
+  }
+
+  // Add the starting and end boundary
+  if (!_loop)
+  {
+    BoundaryInfo & bi = mesh.get_boundary_info();
+    std::vector<BoundaryName> bdy_names{_start_boundary, _end_boundary};
```

Comment:
maybe should warn if a user sets these parameters and `_loop == true`

Reply:
there's an error already